src/windows/com.rs

```diff
@@ -178,48 +221,84 @@ impl FromRawHandle for COMPort {
 
 impl io::Read for COMPort {
     fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
-        let mut len: DWORD = 0;
+        let mut read_size = buf.len();
+
+        let bytes_to_read = self.bytes_to_read()? as usize;
+
+        if self.timeout.as_millis() == 0 && bytes_to_read < read_size {
+            read_size = bytes_to_read;
+        }
+        if read_size == 0 {
+            return Ok(0);
+        }
```

Comment:
Hm there is actually case where `ClearCommError` is useful, but in this library code it is probably not the case
It is mentioned here https://github.com/serialport/serialport-rs/blob/main/src/windows/dcb.rs#L54=

`bytes_to_read`, if flag is set to True would be necessary to resume read operations.

In any case it can be removed since we initialize this flag to false, but I'm generally not sure what is best approach here as I took PR https://gitlab.com/susurrus/serialport-rs/-/merge_requests/91 as reference

Reply:
On side note I think maybe we should only do this check if timeout is zero

Reply:
Note that in case of zero timeout, overlapped IO actually performs indefinite awaiting

Reply:
> Note that in case of zero timeout, overlapped IO actually performs indefinite awaiting

Agreed, but in some cases this might actually be the desired behavior. I think with `Option<Duration>` (see: https://github.com/serialport/serialport-rs/pull/50) we could better express 'never timeout' vs 'never block':

- `None`: Wait until `buf.len()` bytes have been read
- `Some(0)`: Read any pending data, up-to `buf.len()`. Return immediately if there is no data.
- `Some(x)`: Read until `buf.len()` bytes have been returned, or `timeout` has expired

So `None` and `Some(x)` would be deferred to setting `COMMTIMEOUTS`, while `Some(0)` would rely on `bytes_to_read`. I don't know if this is a perfect solution, but I think that `serialport-rs` should be able to support the 'never timeout' case.